lxd/storage/drivers/driver_common.go

```diff
@@ -438,6 +439,38 @@ func (d *common) VolumeSnapshots(vol Volume, op *operations.Operation) ([]string
 	return nil, ErrNotSupported
 }
 
+// HasVolumeSnapshots checks that the volume's snapshots, according to the storage driver, match those provided.
+func (d *common) HasVolumeSnapshots(vol Volume, snapVols []Volume, op *operations.Operation) (bool, error) {
+	// Use the volume's driver reference to pick the actual method as implemented by the driver.
+	storageSnapshotNames, err := vol.driver.VolumeSnapshots(vol, op)
+	if err != nil {
+		return false, err
+	}
+
+	// Create a list of all wanted snapshots.
+	var wantedSnapshotNames []string
```

Comment:
the capacity can be set to len(snapVols)?

Reply:
Thanks, fixed.